frontend/src/components/profile/ProfileCard.jsx

```diff
@@ -21,12 +26,82 @@ export default function ProfileCard({ thisUser, isLoggedInUser }) {
     }
   }
 
+  async function uploadImage(event) {
+    console.log("Sending");
+    event.preventDefault();
+    try {
+      const formData = new FormData();
+      formData.append("file", selectedFile);
+      await UserApi.uploadImage(thisUser.id, formData);
+    } catch (e) {
+      console.error(e);
+    }
+    window.location.reload();
+  }
+
+  const setFile = (event) => {
+    setIsFileSelected(false);
+    const file = event.target.files[0];
+    if (getFileSizeInMB(file.size) > 10) {
+      alert("Files larger than 10MB are not allowed!");
+    } else if (file.size === 0) {
+      alert("Empty files are not allowed!");
+    } else {
+      setSelectedFile(file);
+      setIsFileSelected(true);
+    }
+  };
+
+  function getNiceName(requiredLength) {
+    if (requiredLength >= selectedFile.name.length) {
+      return selectedFile.name;
+    } else {
+      return selectedFile.name.substring(0, requiredLength) + "..";
+    }
+  }
+
   return (
     <div className="ProfileCard">
       <div className="user-fields">
         <div className="name-pair">
-          <img className="user-avatar" src={userURL} alt="User" />
+          {!hasImage && (
+            <img className="user-avatar" src={userURL} alt="User" />
+          )}
+          {hasImage && (
+            <img
+              className="user-image"
+              src={`data:${thisUser.imageType};base64, ${thisUser.image}`}
+              alt="User"
+            />
+          )}
           <h2 className="name">{thisUser.name}</h2>
+
+          {isLoggedInUser && (
+            <div className="profile-image-buttons">
+              <span role="img" aria-label="file pin">
+                📎 {isFileSelected ? getNiceName(16) : "no file selected"}
```

Comment:
Could you cross check with Formfooter and make it so that the line 82 is only visible when file is selected

Reply:
Good suggestion, please check new screenshot added (second pic)